modules/dnn/src/dnn_utils.cpp

```diff
@@ -382,6 +382,112 @@ void imagesFromBlob(const cv::Mat& blob_, OutputArrayOfArrays images_)
+            int rh = int(imgSize.height * resizeFactor);
+            int rw = int(imgSize.width * resizeFactor);
+
+            int top = (size.height - rh) / 2;
+            int left = (size.width - rw) / 2;
+            for (int i = 0; i < rBlob.size(); i++)
+            {
+                rImg[i] = Rect((rBlob[i].x - left) / resizeFactor, (rBlob[i].y - top) / resizeFactor, rBlob[i].width / resizeFactor, rBlob[i].height / resizeFactor);
+            }
+        }
+        else if(this->paddingmode == DNN_PMODE_NULL)
+        {
+            for (int i = 0; i < rBlob.size(); i++)
+            {
+                rImg[i] = Rect(rBlob[i].x * (float)imgSize.width / size.width, rBlob[i].y * (float)imgSize.height / size.height,
+                    rBlob[i].width * (float)imgSize.width / size.width, rBlob[i].height * (float)imgSize.height / size.height);
+            }
+        }
+        else
+            CV_Error(CV_StsBadArg, "Unknown padding mode");
+    }
+}
+
+void Image2BlobParams::blobRectsToImageRects(const std::vector<Rect2d> &rBlob, std::vector<Rect2d>& rImg, const Size& imgSize)
+{
+    Size size = this->size;
+    rImg.resize(rBlob.size());
+    if (size != imgSize)
+    {
+        if (this->paddingmode == DNN_PMODE_CROP_CENTER)
+        {
+            float resizeFactor = std::max(size.width / (float)imgSize.width,
+                size.height / (float)imgSize.height);
+            for (int i = 0; i < rBlob.size(); i++)
+            {
+                rImg[i] = Rect2d((rBlob[i].x + 0.5 * (imgSize.width * resizeFactor - size.width)) / resizeFactor, (rBlob[i].y + 0.5 * (imgSize.height * resizeFactor - size.height)) / resizeFactor,
+                    rBlob[i].width / resizeFactor, rBlob[i].height / resizeFactor);
+            }
+        }
+        else if (this->paddingmode == DNN_PMODE_LETTERBOX)
+        {
+            float resizeFactor = std::min(size.width / (float)imgSize.width,
+                size.height / (float)imgSize.height);
+            int rh = int(imgSize.height * resizeFactor);
+            int rw = int(imgSize.width * resizeFactor);
+
+            int top = (size.height - rh) / 2;
+            int left = (size.width - rw) / 2;
+            for (int i = 0; i < rBlob.size(); i++)
+            {
+                rImg[i] = Rect2d((rBlob[i].x - left) / resizeFactor, (rBlob[i].y - top) / resizeFactor, rBlob[i].width / resizeFactor, rBlob[i].height / resizeFactor);
+            }
+        }
+        else if (this->paddingmode == DNN_PMODE_NULL)
+        {
+            for (int i = 0; i < rBlob.size(); i++)
+            {
+                rImg[i] = Rect2d(rBlob[i].x * (float)imgSize.width / size.width, rBlob[i].y * (float)imgSize.height / size.height,
+                    rBlob[i].width * (float)imgSize.width / size.width, rBlob[i].height * (float)imgSize.height / size.height);
+            }
+        }
+        else
+            CV_Error(CV_StsBadArg, "Unknown padding mode");
+    }
+}
```

Comment:
If the only difference between two methods `Rect` and `Rect2d` in rounding - I recommend to add a note to docstring and reduce code duplication by calling `Rect2d` version and manual conversion to `Rect`